File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/SiteSetting.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 站点设置实体
 * 
 * 采用 Key-Value 存储模式，支持多种数据类型。
 * 
 * @ref §10 - 站点设置表设计
 */
@Data
@Entity
@Table(name = "site_settings", indexes = {
    @Index(name = "idx_site_settings_key", columnList = "setting_key"),
    @Index(name = "idx_site_settings_group", columnList = "group_name")
})
public class SiteSetting {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 设置键名（唯一）
     */
    @Column(name = "setting_key", nullable = false, unique = true, length = 100)
    private String settingKey;

    /**
     * 设置值
     */
    @Lob
    @Column(name = "setting_value", columnDefinition = "TEXT")
    private String settingValue;

    /**
     * 值类型: STRING, NUMBER, BOOLEAN, JSON, TEXT
     */
    @Column(name = "setting_type", nullable = false, length = 20)
    @Enumerated(EnumType.STRING)
    private SettingType settingType = SettingType.STRING;

    /**
     * 分组名称
     */
    @Column(name = "group_name", nullable = false, length = 50)
    private String groupName = "general";

    /**
     * 设置描述
     */
    @Column(length = 500)
    private String description;


    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    /**
     * 设置值类型枚举
     */
    public enum SettingType {
        STRING, NUMBER, BOOLEAN, JSON, TEXT
    }
}

File: apps/server-go/internal/model/site_setting.go
package model

import "time"

// SiteSetting 对应数据库 `site_settings` 表，以键值对形式存储站点全局配置项。
// 配置项按分组（GroupName）归类，管理后台可在不修改代码的情况下动态调整站点选项。
type SiteSetting struct {
	ID           int64     `db:"id"`
	SettingKey   string    `db:"setting_key"`   // 唯一的点分隔配置键（如 "site.title"）
	SettingValue *string   `db:"setting_value"` // 以字符串形式存储的配置值；实际含义由 SettingType 决定
	SettingType  string    `db:"setting_type"`  // 值类型提示：STRING | NUMBER | BOOLEAN | JSON | TEXT
	GroupName    string    `db:"group_name"`    // UI 展示用的逻辑分组名（如 "basic"、"seo"）
	Description  *string   `db:"description"`   // 管理面板中展示的人类可读说明，可为空
	CreatedAt    time.Time `db:"created_at"`    // 记录创建时间
	UpdatedAt    time.Time `db:"updated_at"`    // 记录最后更新时间
}
